internal/v1/api.yaml

```diff
@@ -2031,6 +2036,42 @@ components:
           type: boolean
         module_hotfixes:
           type: boolean
+    CustomRepositoryExported:
+      type: object
+      properties:
+        distribution_arch:
+          description: "Architecture to restrict client usage to"
+          example: "x86_64"
+          type: string
+        distribution_versions:
+          description: "Versions to restrict client usage to"
+          example: [
+            "8"
+          ]
+          type: array
+          items:
+            type: string
+        gpg_key:
+          description: "GPG key for repository"
+          type: string
+        metadata_verification:
+          description: "Verify packages"
+          type: boolean
+        module_hotfixes:
+          description: "Disable modularity filtering on this repository"
+          type: boolean
+        name:
+          description: "Name of the remote yum repository"
+          type: string
+        origin:
+          description: "Origin of the repository"
+          type: string
+        snapshot:
+          description: "Enable snapshotting and hosting of this repository"
+          type: boolean
+        url:
+          description: "URL of the remote yum repository"
+          type: string
```

Comment:
I wonder if image builder should store this in JSON blob simply not caring about the contents. Since the purpose of this is simply being able to recreate the same repository in a different organization, we could simply store version and JSON blob and as long as CS import/export endpoint handles the data correctly we could simply pass it in and out.

Take this as food for thought, this is fine tho we will have to maintain this here.

Reply:
I think this is a relevant point, and we could possibly do that too. This could be easier even in cases when CS decides to change the structure of the ApiRepositoryExportResponse endpoint. However, the most readable way to send a request is again initializing an ApiRepositoryExportRequest struct, and passing in all the data. So I think we would have to unmarshal the JSON anyway.

Reply:
Fair enough.